pkg/controllers/provisioning/provisioner.go

```diff
@@ -225,6 +227,10 @@ func (p *Provisioner) NewScheduler(ctx context.Context, pods []*v1.Pod, stateNod
 	instanceTypes := map[string][]*cloudprovider.InstanceType{}
 	domains := map[string]sets.Set[string]{}
 	for _, nodePool := range nodePoolList.Items {
+		if !nodePool.StatusConditions().IsTrue(status.ConditionReady) {
```

Comment:
Do we have a provisioning test somewhere to ensure that we don't use NotReady NodePools?